### PR TITLE
Enable create settings to access resource attributes

### PR DESCRIPTION
**Pull request by jaronoff97**
Adds a ResourceAttrs field to `component.TelemetrySettings` which exists in each `CreateSettings` object. This means that a component will be access instantiated resource attributes upon pipeline creation.

### CODE REVIEW

File: service/service.go [hunk 1]
```diff
@@ -19,6 +19,9 @@ import (
 	"fmt"
 	"runtime"
 
+	"github.com/google/uuid"
+	semconv "go.opentelemetry.io/collector/semconv/v1.5.0"
```
Comment (paivagustavo) [inline service/service.go:23]: need to sort this import :)

File: extension/extension_test.go [hunk 2]
```diff
@@ -113,6 +113,11 @@ func TestBuilder(t *testing.T) {
 	assert.NoError(t, err)
 	assert.NotNil(t, e)
 
+	// Check that the extension has access to the resource attributes.
+	nop, ok := e.(nopExtension)
+	assert.True(t, ok)
+	assert.Len(t, nop.CreateSettings.ResourceAttrs, 0)
```
Comment (paivagustavo) [inline extension/extension_test.go:119]: Should we also have a test where `ResourceAttrs` is not empty?

**Review by paivagustavo — APPROVED** [review]
LGTM, might just add another test case but other than that seems fine!

File: extension/extension_test.go (hunk 2 above)
Comment (jaronoff97) [reply extension/extension_test.go:119]: In this case we always expect it to be empty here because it's not being instantiated by a service

Comment (jaronoff97) [reply extension/extension_test.go:119]: we already have extensive tests for the service and getting the resources from there... maybe i'll edit one of them to ensure the expected resources exist in the telemetry settings.

**Review by paivagustavo — APPROVED** [review]
